`update-lingering` activation script is broken

https://github.com/NixOS/nixpkgs/pull/260248#discussion_r1428504292

```
[ERROR]   stderr) Could not enable linger: Connection timed out
[ERROR]   stdout) Activation script snippet 'update-lingering' failed (123)
[ERROR]   stderr) Failed to list users: Connection timed out
[ERROR]   stderr) Unable to close the file handle to loginctl at /nix/store/hiw28n1i7mmr7sh7ds66g9nmxdbf7kdv-nixos-system-cetus-23.11pre-git/bin/switch-to-configuration line 890.
```

nixpkgs input: `github:NixOS/nixpkgs/093cffd33371b29257a4302206c214e87a0297ac`

cc @ToxicFrog @amaxine

--- a/nixos/modules/config/users-groups.nix
+++ b/nixos/modules/config/users-groups.nix
@@ -704,6 +704,11 @@ in {
     in stringAfter [ "users" ] ''
       if [ -e ${lingerDir} ] ; then
         cd ${lingerDir}
+        for user in ${lingerDir}/*; do
+          if ! id "$user" >/dev/null 2>&1; then
+            rm --force -- "$user"
+          fi
+        done
         ls ${lingerDir} | sort | comm -3 -1 ${lingeringUsersFile} - | xargs -r ${pkgs.systemd}/bin/loginctl disable-linger
         ls ${lingerDir} | sort | comm -3 -2 ${lingeringUsersFile} - | xargs -r ${pkgs.systemd}/bin/loginctl  enable-linger
       fi

